pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -273,6 +273,24 @@ protected CompletableFuture<Map<String, Set<AuthAction>>> internalGetPermissions
                         }
                     }
                 }
+
+                // If topic is partitioned, add based topic permission
```

Comment:
I do not think we should grant permission based on the partition's base topic. If the call is about a specific permission, we should respond with the result for that partition.

Reply:
@michaeljmarshall Is your idea the same as I described in  https://github.com/apache/pulsar/pull/16792#issuecomment-1219114676 ?? That's what I thought at the beginning, marked as plan 1.

plan 2 is described in https://github.com/apache/pulsar/pull/16792#issue-1317797371, this one is designed based on the idea of @nodece .

Both of the two plan can enhance.  I think plan 1 would have better compatibility with previous version, while plan 2 would have cleaner permission record.